src/debug_info.rs

```diff
@@ -1,30 +1,475 @@
+                if is_optimized { 0 } else { 1 },
+            )
+        };
+        DISubProgram {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+        /*
+        pub unsafe extern "C" fn LLVMDIBuilderCreateFunction(
+            Builder: LLVMDIBuilderRef,
+            Scope: LLVMMetadataRef,
+            Name: *const c_char,
+            NameLen: size_t,
+            LinkageName: *const c_char,
+            LinkageNameLen: size_t,
+            File: LLVMMetadataRef,
+            LineNo: c_uint,
+            Ty: LLVMMetadataRef,
+            IsLocalToUnit: LLVMBool,
+            IsDefinition: LLVMBool,
+            ScopeLine: c_uint,
+            Flags: LLVMDIFlags,
+            IsOptimized: LLVMBool
+        ) -> LLVMMetadataRef
+         *
+         */
+    }
+
+    pub fn create_file(&self, filename: &str, directory: &str) -> DIFile<'ctx> {
+        let filename = CString::new(filename).expect("Conversion to CString failed unexpectedly");
+        let directory = CString::new(directory).expect("Conversion to CString failed unexpectedly");
+        let metadata_ref = unsafe {
+            LLVMDIBuilderCreateFile(
+                self.builder,
+                filename.as_ptr(),
+                filename.as_bytes().len(),
+                directory.as_ptr(),
+                directory.as_bytes().len(),
+            )
+        };
+        DIFile {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn create_debug_location(
+        &self,
+        context: &Context,
+        line: u32,
+        column: u32,
+        scope: DIScope<'ctx>,
+    ) -> DebugLocation<'ctx> {
+        let metadata_ref = unsafe {
+            LLVMDIBuilderCreateDebugLocation(
+                context.context,
+                line,
+                column,
+                scope.metadata_ref,
+                std::ptr::null_mut(),
+            )
+        };
+        /*
+        LLVMDIBuilderCreateDebugLocation(
+            Ctx: LLVMContextRef,
+            Line: c_uint,
+            Column: c_uint,
+            Scope: LLVMMetadataRef,
+            InlinedAt: LLVMMetadataRef
+        ) -> LLVMMetadataRef
+            */
+        DebugLocation {
+            metadata_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    /// * `encoding` -
+    /// ```ignore
+    /// const DW_LANG_RUST: c_uint = 0x1c;
+    /// #[allow(non_upper_case_globals)]
+    /// const DW_ATE_boolean: c_uint = 0x02;
+    /// #[allow(non_upper_case_globals)]
+    /// const DW_ATE_float: c_uint = 0x04;
+    /// #[allow(non_upper_case_globals)]
+    /// const DW_ATE_signed: c_uint = 0x05;
+    /// #[allow(non_upper_case_globals)]
+    /// const DW_ATE_unsigned: c_uint = 0x07;
+    /// #[allow(non_upper_case_globals)]
+    /// const DW_ATE_unsigned_char: c_uint = 0x08;
+    /// ```
+    pub fn create_basic_type(
+        &self,
+        name: &str,
+        size_in_bits: u64,
+        encoding: LLVMDWARFTypeEncoding,
```

Comment:
You're going to need to create a new wrapper enum for this. See https://github.com/TheDan64/inkwell/blob/9dd5214a1c5ef5b8a9985847624eaf10c535af9e/src/values/global_value.rs#L336-L350 for example

Reply:
For this and every other LLVM flag re-export, right?